test-cypress/plugins/index.js

```diff
@@ -26,4 +28,18 @@ module.exports = (on, config) => {
             'cypress-logs|txt': 'txt'
         }
     });
+
+    // keep only the videos for the failed specs
+    on('after:spec', (spec, results) => {
+        if (results && results.video) {
+            // Do we have failures for any retry attempts?
+            const failures = _.some(results.tests, (test) => {
```

Comment:
You actually don't need lodash for this. You can rewrite it something like this
```javascript
const failures = results.tests.find((test) => {
  return test.attempts.find((attempt) => attempt.state === 'failed');
});
```

Reply:
Used JS stream functions.